azurerm/resource_arm_azure_firewall_network_rule_collection.go

```diff
@@ -0,0 +1,342 @@
+		return nil
+	}
+
+	d.Set("name", collection.Name)
+	d.Set("action", string(collection.AzureFirewallNetworkRuleCollectionPropertiesFormat.Action.Type))
+	d.Set("priority", collection.AzureFirewallNetworkRuleCollectionPropertiesFormat.Priority)
+	if rules := collection.AzureFirewallNetworkRuleCollectionPropertiesFormat.Rules; rules != nil {
+		flattenedRules := flattenArmAzureFirewallNetworkRules(rules)
+		if err := d.Set("rule", flattenedRules); err != nil {
+			return fmt.Errorf("Error setting `rule`: %+v", err)
+		}
+	}
+
+	return nil
+}
+
+func resourceArmAzureFirewallNetworkRuleCollectionDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).azureFirewallsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	name := d.Get("name").(string)
+	resourceGroup := id.ResourceGroup
+	firewallName := id.Path["azureFirewalls"]
+
+	azureRMLockByName(firewallName, azureFirewallResourceName)
+	defer azureRMUnlockByName(firewallName, azureFirewallResourceName)
+
+	firewall, err := client.Get(ctx, resourceGroup, firewallName)
+	if err != nil {
+		return fmt.Errorf("Error making Read request on Azure Firewall %q (Resource Group %q): %+v", firewallName, resourceGroup, err)
+	}
+	_, _, exists := findArmAzureFirewallNetworkRuleCollectionByName(&firewall, name)
+	if !exists {
+		return nil
+	}
+	updatedCollections := removeArmAzureFirewallNetworkRuleCollectionByName(&firewall, name)
+	firewall.AzureFirewallPropertiesFormat.NetworkRuleCollections = updatedCollections
+
+	ipConfigs := fixArmAzureFirewallIPConfiguration(&firewall)
+	firewall.AzureFirewallPropertiesFormat.IPConfigurations = &ipConfigs
+
+	future, err := client.CreateOrUpdate(ctx, resourceGroup, firewallName, firewall)
+	if err != nil {
+		return fmt.Errorf("Error creating/updating Azure Firewall %q (Resource Group %q): %+v", firewallName, resourceGroup, err)
+	}
+
+	err = future.WaitForCompletion(ctx, client.Client)
+	if err != nil {
+		return fmt.Errorf("Error waiting for creation/update of Azure Firewall %q (Resource Group %q): %+v", firewallName, resourceGroup, err)
+	}
+
+	return nil
+}
+
+func expandArmAzureFirewallNetworkRuleCollection(d *schema.ResourceData) network.AzureFirewallNetworkRuleCollection {
+	name := d.Get("name").(string)
+	action := network.AzureFirewallRCActionType(d.Get("action").(string))
+	rules := expandArmAzureFirewallNetworkRules(d)
+	properties := network.AzureFirewallNetworkRuleCollectionPropertiesFormat{
+		Action: &network.AzureFirewallRCAction{
+			Type: action,
+		},
+		Priority: utils.Int32(int32(d.Get("priority").(int))),
+		Rules:    &rules,
+	}
+	col := network.AzureFirewallNetworkRuleCollection{
+		Name: &name,
+		AzureFirewallNetworkRuleCollectionPropertiesFormat: &properties,
+	}
+	return col
+}
+
+func findArmAzureFirewallNetworkRuleCollectionByName(firewall *network.AzureFirewall, name string) (*network.AzureFirewallNetworkRuleCollection, int, bool) {
```

Comment:
we should be able to make the int a pointer here, and return nil when it doesn't exist?